deployment/ccip/changeset/cs_rmn_curse_uncurse.go

```diff
@@ -0,0 +1,275 @@
+package changeset
+
+import (
+	"encoding/binary"
+	"fmt"
+	"slices"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	commoncs "github.com/smartcontractkit/chainlink/deployment/common/changeset"
+)
+
+func GlobalCurseSubject() Subject {
+	return Subject{0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01}
+}
+
+type RMNCurseAction struct {
+	ChainSelector  uint64
+	SubjectToCurse Subject
+}
+
+type CurseAction func(e deployment.Environment) []RMNCurseAction
+
+type RMNCurseConfig struct {
+	MCMS         *MCMSConfig
+	CurseActions []CurseAction
+	Reason       string
+}
+
+func (c RMNCurseConfig) Validate(e deployment.Environment) error {
+	state, err := LoadOnchainState(e)
+
+	if err != nil {
+		return errors.Errorf("failed to load onchain state: %v", err)
+	}
+
+	if len(c.CurseActions) == 0 {
+		return errors.Errorf("curse actions are required")
+	}
+
+	if c.Reason == "" {
+		return errors.Errorf("reason is required")
+	}
+
+	validSelectors := e.AllChainSelectors()
+	validSubjects := make([]Subject, 0, len(validSelectors)+1)
+	for _, selector := range validSelectors {
+		validSubjects = append(validSubjects, SelectorToSubject(selector))
+	}
+	validSubjects = append(validSubjects, GlobalCurseSubject())
```

Comment:
This works fine but in Go, `map`s are used to check for inclusion since its a constant time look up. In this case you want basically a set, so `map[Subject]struct{}` would work fine.

```golang
validSubjects := map[Subject]struct{}{
  GlobalCurseSubject(): struct{}{},
}
for _, selector := range validSelectors {
  validSubjects[SelectorToSubject(selector)] = struct{}{}
}

// later down in the code
if _, ok := validSubjects[action.SubjectToCurse]; !ok {
  return fmt.Errorf("invalid subject %x for chain %s", action.SubjectToCurse, targetChain)
}
```